% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/dada.R
\name{setDadaOpt}
\alias{setDadaOpt}
\title{Set DADA options}
\usage{
setDadaOpt(...)
}
\arguments{
\item{...}{(Required). The DADA options to set, along with their new value.}
}
\value{
NULL.
}
\description{
setDadaOpt sets the default options used by the dada(...) function for your current session, much
 like \code{par} sets the session default plotting parameters. However, all dada options can be set as
 part of the dada(...) function call itself by including a DADA_OPTION_NAME=VALUE argument.
}
\details{
The various dada options...

OMEGA_A: This parameter sets the threshold for when DADA2 calls unique sequences significantly overabundant, and therefore creates a
 new cluster with that sequence as the center. Default is 1e-40, which is a conservative setting to avoid making false
 positive inferences, but which comes at the cost of reducing the ability to identify some rare variants.

USE_QUALS: If TRUE, the dada(...) error model takes into account the consensus quality score of the dereplicated unique sequences.
 If FALSE, quality scores are ignored. Default is TRUE.

USE_KMERS: If TRUE, a 5-mer distance screen is performed prior to performing each pairwise alignment, and if the 5mer-distance
 is greater than KDIST_CUTOFF, no alignment is performed. Default is TRUE.

KDIST_CUTOFF: The default value of 0.42 was chosen to screen pairs of sequences that differ by >10\%, and was
 calibrated on Illumina sequenced 16S amplicon data. The assumption is that sequences that differ by such a large
 amount cannot be linked by amplicon errors (i.e. if you sequence one, you won't get a read of other) and so
 careful (and costly) alignment is unnecessary.

BAND_SIZE: When set, banded Needleman-Wunsch alignments are performed. Banding restricts the net cumulative number of insertion
 of one sequence relative to the other. The default value of BAND_SIZE is 16. If DADA is applied to marker genes with high rates
 of indels, such as the ITS region in fungi, the BAND_SIZE parameter should be increased. Setting BAND_SIZE to a negative number
 turns off banding (i.e. full Needleman-Wunsch).

SCORE_MATRIX: The score matrix for the Needleman-Wunsch alignment. This is a 4x4 matrix as no ambiguous nucleotides
 are allowed. Default is nuc44: -4 for mismatches, +5 for matches.
 
GAP_PENALTY: The cost of gaps in the Needleman-Wunsch alignment. Default is -8.
 
HOMOPOLYMER_GAP_PENALTY: The cost of gaps in homopolymer regions (>=3 repeated bases). Default is NULL, which causes homopolymer
 gaps to be treated as normal gaps.  
 
MIN_FOLD: The minimum fold-overabundance for sequences to form new clusters. Default value is 1, which means this
 criteria is ignored.
 
MIN_HAMMING: The minimum hamming-separation for sequences to form new clusters. Default value is 1, which means this
 criteria is ignored.

MAX_CLUST: The maximum number of clusters. Once this many clusters have been created, the algorithm terminates regardless
 of whether the statistical model suggests more real sequence variants exist. If set to 0 this argument is ignored. Default
 value is 0.
 
MAX_CONSIST: The maximum number of steps when selfConsist=TRUE. If convergence is not reached in MAX_CONSIST steps,
 the algorithm will terminate with a warning message. Default value is 10.
 
VERBOSE: If TRUE progress messages from the algorithm are printed. Warning: There is a lot of output. Default is FALSE.
}
\examples{
setDadaOpt(OMEGA_A = 1e-20)
setDadaOpt(OMEGA_A = 1e-20, VERBOSE = TRUE)
  
}
\seealso{
\code{\link{getDadaOpt}}
}
